### PR TITLE
Amend the submission schema to handle the updated Census downstream submission data

### PR DESCRIPTION
**Pull request by berroar**
# Context
The 'data' field within the Census Downstream Submission Data has been amended in order to handle list collection - [See the PR on Runner](https://github.com/ONSdigital/eq-survey-runner/pull/2294).

Census previously sent the 'data' field in the following format:
```python
data: [{
  answer_id: 'test'
  list_item_id: '123'
  value: 'testing'
}]
```

This has been amended to be:
```python
data: {
    answers: [{
    answer_id: 'test'
    list_item_id: '123'
    value: 'testing'
  }],
  lists: [{
    name: 'people',
    items: ['123', 'abc']
  }]
}
```
As a result we have updated schemas/eq_submission_schema.json so that it validates the updated 'data' field correctly.

We have also added the test_relationships_response.json file as an example Census output file.

### CODE REVIEW

**Comment by berroar** [comment]
[Sending census submission data downstream (M)](https://trello.com/c/ed4j3mgZ/2963-sending-census-submission-data-downstream-m)

File: schemas/eq_submission_schema.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
   "definitions": {},
   "$schema": "http://json-schema.org/draft-07/schema#",
-  "$id": "https://edc.ons.gov.uk/schemas/eq_submission.json",
+  "$id": "http://example.com/root.json",
```
Comment (collisdigital) [inline schemas/eq_submission_schema.json]: Looks like these changes shouldn't be here

File: schemas/eq_submission_schema.json [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
   "type": "object",
-  "title": "ONS EQ Submission Schema",
+  "title": "The Root Schema",
```
Comment (collisdigital) [inline schemas/eq_submission_schema.json]: Looks like these changes shouldn't be here

File: schemas/eq_submission_schema.json [hunk 3]
```diff
@@ -16,83 +16,79 @@
...
     "started_at",
+    "case_id",
     "data"
   ],
   "properties": {
     "tx_id": {
       "$id": "#/properties/tx_id",
       "type": "string",
-      "title": "Transaction ID",
-      "description": "Used to trace a transaction through the whole system",
```
Comment (collisdigital) [inline schemas/eq_submission_schema.json]: are we removing these on purpose?

File: schemas/eq_submission_schema.json [hunk 4]
```diff
@@ -16,83 +16,79 @@
...
+      "title": "The Tx_id Schema",
       "default": "",
       "examples": [
-        "f81d4fae-7dec-11d0-a765-00a0c91e6bf6"
+        "411e12a2-e57f-4f07-917d-76c93b422460"
       ],
-      "pattern": "^[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}$"
```
Comment (collisdigital) [inline schemas/eq_submission_schema.json]: this was more specific before this change, why the change?

File: schemas/eq_submission_schema.json [hunk 5]
```diff
@@ -16,83 +16,79 @@
...
+      "pattern": "^(.*)$"
     },
     "type": {
       "$id": "#/properties/type",
       "type": "string",
-      "title": "Type",
-      "description": "The unique type identifier of this JSON file. Can be \"uk.gov.ons.edc.eq:surveyresponse\" or \"uk.gov.ons.edc.eq:feedback\"",
```
Comment (collisdigital) [inline schemas/eq_submission_schema.json]: Same as above, review these deletions throughout...

File: schemas/test_relationships_response.json [hunk 6]
```diff
@@ -0,0 +1,71 @@
+{
```
Comment (collisdigital) [inline schemas/test_relationships_response.json]: Should this be in the examples/ folder and/or should it replace https://github.com/ONSdigital/ons-schema-definitions/blob/v3/examples/eq_submission_example.json ?

File: schemas/eq_submission_schema.json (hunk 5 above)
Comment (berroar) [reply schemas/eq_submission_schema.json]: We've now reverted the changes that we had inadvertently overwrote.

File: schemas/test_relationships_response.json (hunk 6 above)
Comment (berroar) [reply schemas/test_relationships_response.json]: We refreshed the eq_submission_example schema and have now removed the test_relationships_schema.

File: schemas/eq_submission_schema.json [hunk 7]
```diff
@@ -187,7 +190,27 @@
...
+      ],
+      "pattern": "^(.*)$"
+    },
+    "case_type": {
+      "$id": "#/properties/case_type",
+      "type": "string",
+      "title": "The Case_type Schema",
+      "default": "",
+      "examples": [
+        "HI"
```
Comment (bitdivision) [inline schemas/eq_submission_schema.json]: This field should be an enum. 

Runner validates it as: `validate=validate.OneOf(('HH', 'HI', 'CE', 'CI'))

File: schemas/eq_submission_schema.json [hunk 8]
```diff
@@ -187,7 +190,27 @@
...
+      ],
+      "pattern": "^(.*)$"
+    },
+    "region_code": {
+      "$id": "#/properties/region_code",
+      "type": "string",
```
Comment (bitdivision) [inline schemas/eq_submission_schema.json]: Would be useful to add a description showing that this should be `A region code defined as per ISO 3166-2:GB`

Comment (bitdivision) [reply schemas/eq_submission_schema.json]: You could even include the regex from runner if you felt like it

File: schemas/eq_submission_schema.json [hunk 9]
```diff
@@ -197,22 +197,27 @@
     "case_type": {
       "$id": "#/properties/case_type",
       "type": "string",
+      "items": {
+        "type": "array",
```
Comment (bitdivision) [inline schemas/eq_submission_schema.json]: Runner does  not accept an array at the moment?

I think this should be type string?

File: schemas/eq_submission_schema.json [hunk 10]
```diff
@@ -197,22 +197,27 @@
...
+        "enum": ["HH", "HI", "CE", "CI"]
+      },
       "title": "The Case_type Schema",
       "default": "",
       "examples": [
         "HI"
       ],
-      "pattern": "^(.*)$"
+      "pattern": "^[A-Z]{2}$"
```
Comment (bitdivision) [inline schemas/eq_submission_schema.json]: This regex isn't necessary, since the values are now enumerated.

Comment (petechd) [reply schemas/eq_submission_schema.json]: Can you check these fixes I've made please?

File: schemas/eq_submission_schema.json (hunk 9 above)
Comment (petechd) [reply schemas/eq_submission_schema.json]: I've made requested changes, thanks.

File: schemas/eq_submission_schema.json [hunk 11]
```diff
@@ -198,15 +198,15 @@
       "$id": "#/properties/case_type",
       "type": "string",
       "items": {
```
Comment (bitdivision) [inline schemas/eq_submission_schema.json]: Isn't this for an array @petechd ?

File: schemas/eq_submission_schema.json [hunk 12]
```diff
@@ -187,18 +190,40 @@
...
+    "case_type": {
+      "$id": "#/properties/case_type",
+      "type": "string",
+      "enum": ["HH", "HI", "CE", "CI"],
+      "title": "The Case_type Schema",
+      "default": "",
+      "examples": [
+        "HI"
+      ],
+      "pattern": "^(.*)$"
```
Comment (bitdivision) [inline schemas/eq_submission_schema.json]: Can just remove this pattern rather than having a wildcard pattern?

File: schemas/eq_submission_schema.json [hunk 13]
```diff
@@ -208,13 +233,13 @@
...
       "examples": [
-        "42d78898-7e7a-40b9-9500-757bb546e358"
+        "5515cd5f-fcff-44f0-ac02-3530477693ea"
       ],
       "pattern": "^[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}$"
     },
     "data": {
       "$id": "#/properties/data",
-      "type": "array",
+      "type":  "object",
```
Comment (bitdivision) [inline schemas/eq_submission_schema.json]: Shouldn't this now have `data` and `lists` properties?

Comment (petechd) [reply schemas/eq_submission_schema.json]: We've incorporated lists to data section now, can you check please?

File: schemas/eq_submission_schema.json [hunk 14]
```diff
@@ -239,59 +238,120 @@
...
+                  "first-name",
+                  "ethnic-group-answer"
+                ],
+                "pattern": "^(.*)$"
+              },
+              "value": {
+                "$id": "#/properties/data/properties/answers/items/properties/value",
+                "type": ["string", "array"],
+                "title": "The response value",
+                "description": "Can be a single string or multiple strings (e.g. for checkboxes)",
```
Comment (bitdivision) [inline schemas/eq_submission_schema.json]: Will this still validate multiple items? e.g. for an array, now that items has been removed I don't think this will validate?

Comment (petechd) [reply schemas/eq_submission_schema.json]: Added items to "value" section, thanks for help!
